erizo/src/erizo/IceConnection.h

```diff
@@ -0,0 +1,132 @@
+/*
+ * NiceConnection.h
+ */
+
+#ifndef ERIZO_SRC_ERIZO_ICECONNECTION_H_
+#define ERIZO_SRC_ERIZO_ICECONNECTION_H_
+
+#include <boost/scoped_ptr.hpp>
+#include <boost/thread.hpp>
+#include <string>
+#include <vector>
+#include <queue>
+#include <map>
+
+#include "./MediaDefinitions.h"
+#include "./SdpInfo.h"
+#include "./logger.h"
+#include "lib/LibNiceInterface.h"
+
+typedef struct _NiceAgent NiceAgent;
+typedef struct _GMainContext GMainContext;
+typedef struct _GMainLoop GMainLoop;
+
+typedef unsigned int uint;
+
+namespace erizo {
+
+// forward declarations
+typedef std::shared_ptr<dataPacket> packetPtr;
+class CandidateInfo;
+class WebRtcConnection;
+class IceConnection;
+
+struct CandidatePair{
+  std::string erizoCandidateIp;
+  int erizoCandidatePort;
+  std::string clientCandidateIp;
+  int clientCandidatePort;
+  std::string erizoHostType;
+  std::string clientHostType;
+};
+
+class IceConfig {
```

Comment:
@lodoyun can you review the new IceConfig? I added media_type transport_name, connection_id, ice_components, username and password and removed them from the NiceConnection constructor.